### DISCUSSION TITLE
What is the best way to review logs and make sure no one abuse my instance?

### DISCUSSION THREAD

**Discussion by XDavidT** [dani-garcia/vaultwarden#3334] [General]
I want to suggest my organization work with VaultWarden, but we had some questions before starting to use them.
We really want to trust it, but the most important thing is the logging in our case.
Are the logs can tell what IPs are been used to log in? or any other way that helps us to prevent abuse on the system?
What is your concern when you first start to use it?

**Comment by RealOrangeOne** [comment]
You probably want a reverse proxy in front, which could absolutely tell you which IPs are logging in. For abuse, the same methods for preventing abuse in any system will apply.

**Reply by XDavidT** [reply]
So assume using nginx controller via K8s cluster, is good enough?
What any other policies are recommended?

**Reply by RealOrangeOne** [reply]
Nginx is a reverse proxy, yes. 

Again, anything you'd do to secure a web application. WAF, rate limiting, HSTS, all things which can be useful. It's best to look up these things for yourself, as they're not vaultwarden-specific.
